src/Backend/Modules/Authentication/Actions/Index.php

```diff
@@ -268,4 +259,73 @@ private function validateForm()
             }
         }
     }
+
+    /*
+     * Find out which module and action are allowed
+     * and send the user on his way.
+     */
+    private function redirectToAllowedModuleAndAction()
+    {
+        $allowedModule = $this->getAllowedModule();
+        $allowedAction = $this->getAllowedAction($allowedModule);
+        $this->redirect($this->getParameter('querystring', 'string',
+            $allowedModule ?
+                BackendModel::createUrlForAction($allowedAction, $allowedModule) :
+                BackendModel::createUrlForAction('index', 'Authentication')
+            )
+        );
+    }
+
+    /*
+     * Run through the action of a certain module
+     * and find us an action(name) this user is
+     * allowed to access.
+     */
+    private function getAllowedAction($module)
+    {
+        if(BackendAuthentication::isAllowedAction('Index', $module)) {
+            return 'Index';
+        }
+        $allowedAction = false;
+        $finder = new Finder();
+        $finder->name('*.php')
```

Comment:
Can't you just get the allowed actions out of the db? It's stored in the groups_rights_actions table.